src/components/Stream.tsx

```diff
@@ -280,12 +282,48 @@ export const Stream = () => {
     }
   }
 
+  /**
+   * On double-click of sketch entities we automatically enter sketch mode with the selected sketch,
+   * allowing for quick editing of sketches. TODO: This should be moved to a more central place.
+   */
+  const enterSketchModeIfSelectingSketch: MouseEventHandler<HTMLDivElement> = (
+    e
+  ) => {
+    if (
+      !isNetworkOkay ||
+      !videoRef.current ||
+      state.matches('Sketch') ||
+      state.matches({ idle: 'showPlanes' }) ||
+      sceneInfra.camControls.wasDragging === true ||
+      !btnName(e.nativeEvent).left
+    )
+      return
+    sendSelectEventToEngine(e, videoRef.current)
+      .then(({ entity_id }) => {
+        if (!entity_id) return // No entity selected. This is benign
+        const artifact = engineCommandManager.artifactGraph.get(entity_id)
+        // If an ID is returned and the artifact is not found, we can't do anything, but that's bad and worth reporting
+        if (!artifact)
+          return reportRejection(`No artifact with ID: ${entity_id}`)
+        const sketchArtifactTypes: Artifact['type'][] = [
+          'path',
+          'solid2D',
+          'segment',
+        ]
+        if (sketchArtifactTypes.includes(artifact.type)) {
+          sceneInfra.modelingSend({ type: 'Enter sketch' })
+        }
+      })
+      .catch(reportRejection)
```

Comment:
Not tested, and maybe `reportRejection` still needs to be mixed in here, but this is a little more concise with existing artifactGraph helpers.
```suggestion
        const artifact = getArtifactOfTypes(
    { key: entity_id, types: ['path', solid2D, 'segment'] },
    engineCommandManager.artifactGraph
  )
  if (err(path)) return path
  sceneInfra.modelingSend({ type: 'Enter sketch' })
```

Reply:
Oo so pretty

Reply:
Accepting commit suggestion then testing/reviewing locally